sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -164,6 +164,9 @@ int hda_dsp_ctrl_clock_power_gating(struct snd_sof_dev *sdev, bool enable)
 int hda_dsp_ctrl_init_chip(struct snd_sof_dev *sdev, bool full_reset)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	struct hdac_ext_link *hlink = NULL;
```

Comment:
Looks like a superfluous initialisation, please, check and - if confirmed - remove

Reply:
@lyakh the code is needed for reset the stream-to-link mapping, both for chip power up and the DSP resume.

Reply:
@lyakh removed that initialisation.